charts/opentelemetry-demo/templates/_pod.tpl

```diff
@@ -41,25 +41,29 @@ Get Services Port Mapping
 {{- end }}
 
 {{/*
-Get Pod Env 
+Get Pod Env
 */}}
 {{- define "otel-demo.pod.env" -}}
 {{- $prefix := include "otel-demo.name" $ }}
 
+{{- if .default  }}
+{{- toYaml .default.env -}}
+{{- end }}
+
 {{- if .env }}
-{{- toYaml .env }}
+{{ toYaml .env }}
 {{- end }}
 
-{{- if .observability.otelcol.enabled }} 
+{{- if .observability.otelcol.enabled }}
 - name: OTEL_EXPORTER_OTLP_ENDPOINT
   value: http://{{ include "otel-demo.name" . }}-otelcol:4317
 - name: OTEL_RESOURCE_ATTRIBUTES
-  value: service.name={{ .name | kebabcase }}
+  value: service.name={{ .name | kebabcase }},k8s.namespace.name=$(OTEL_K8S_NAMESPACE),k8s.node.name=$(OTEL_RESOURCE_ATTRIBUTES_NODE_NAME),k8s.pod.name=$(OTEL_RESOURCE_ATTRIBUTES_POD_NAME)
```

Comment:
I think normally in a k8s deployment using otel these values would be added via the collector.  What is the advantage of using `OTEL_RESOURCE_ATTRIBUTES` instead?

Reply:
sorry, the newest commit have recover this line. 

In my scenario, the collector and actual service is deploying in a different namespace, so the collector doesn't know which namespace the service is deployed or what the pod name which the service's container is running. Pod name is a very important uniform identity for us, in the service monitor scenario, we want to know every service instance's metrics.

Reply:
I guess since the k8sattributes processor requires elevated permissions and downard api doesn't this solution makes sense.

Reply:
Actually, these 3 values are pretty standard for k8s deployments using otel.  Maybe it does make sense to include them by default.

Reply:
@dmitryax @Allex1 @wph95 what do you think?

Reply:
We can start with this approach. It makes sense to me. 

I'd say ideally we should use a fully featured collector installation with `k8sattributes` processor, but we don't even have that in the collector chart. @TylerHelmuth I think that is a good candidate for another preset, WDYT? 

Maybe later we can use the collector with enabled `k8sattributes` processor as a subchart here.

Reply:
@dmitryax now that we've got presets in place I think we could better support k8sattributes and its RBAC needs in the opentelemetry-collector chart.  

I'm not entirely sure we'll want to introduce the same RBAC requirements into the demo chart tho; my gut says the less the demo chart has to install the better.  This PR's approach means we won't need any cluster roles or cluster role bindings.  I'm open to the idea tho.

Reply:
@Frapschen please add back these 3 fields as defaults, but configurable.